dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -122,8 +129,75 @@ def local_workshop_daily_survey_report
       end
     end
 
+    # GET /api/v1/pd/workshops/:id/generic_survey_report
+    def generic_survey_report
+      return local_workshop_daily_survey_report if @workshop.local_summer? || @workshop.teachercon? ||
+        ([COURSE_CSP, COURSE_CSD].include?(@workshop.course) &&
+        @workshop.workshop_starting_date > Date.new(2018, 8, 1))
```

Comment:
Should CSF 202 be included in this check, as you did in the client code?

Reply:
I know it's confusing. This condition is for `local_workshop_daily_survey_report`, it's the same as the one on line 116. 
The conditions in the client code is for `daily_survey_results` UI path, which sends request to this endpoint and cover both local summer, academic-year and CSF workshops. It uses the same view to display results for those workshops.